### DISCUSSION TITLE
Should component output be in OSCAL format?

### DISCUSSION THREAD

**Discussion by ErnestoOrtiz3** [usnistgov/OSCAL#1705] [Q&A] [answered]
Are the output of tools (say a static code analizer) something that would be transformed into OSCAL format, or is OSCAL at a higher level of abstraction, where the implementation and assessment of controls describe how things work without including the results of particular tools?

**Comment by openprivacy** [comment]
I am not endorsing nor do I have much experience with this tool, but it seems the MITRE SAF folk have a take on [normalizing security tool output](https://saf.mitre.org/#/normalize) that may be useful to OSCAL Assessment processes.

**Reply by ErnestoOrtiz3** [reply]
Thank you, I'll take a look at it.

**Comment by nikitawootten-nist** [comment, marked as answer]
The OSCAL [Assessment Results](https://pages.nist.gov/OSCAL/concepts/layer/assessment/assessment-results/) model is composed of (among other things): observations, risks, and findings. [Observations](https://pages.nist.gov/OSCAL/reference/latest/assessment-results/json-reference/#/assessment-results/results/observations) can contain human or machine-generated evidence of compliance or non-compliance.

OSCAL is on the "higher level of abstraction" side, allowing for implementors to collect evidence from a wide variety of tools and processes without prescribing a specific tool output.

For a really simple example of an automated workflow generating Assessment Results documents based off of tool output (in this case, a Python test), check out this [case study](https://github.com/usnistgov/blossom-case-study) that our team recently presented on.

**Reply by ErnestoOrtiz3** [reply]
Thank you! This seems very helpful. I'll work through it, and maybe I'll post a follow up question or comment in a few days.
